### PR TITLE
Fix Skipper exception handling on upgrade

### LINKED ISSUE #997: Updating stream with the same version as the deployed version returns a stack trace
When updating a stream with the same version via dataflow-shell, it returns a SkipperException. Since the was no difference to existing deployed/deleted package, it should return a user message.

```
dataflow:>stream update --name mystream3 --properties "version.log=3.0.1"
Command failed org.springframework.cloud.dataflow.rest.client.DataFlowClientException: SkipperException
SkipperException
org.springframework.cloud.dataflow.rest.client.DataFlowClientException: SkipperException
at org.springframework.cloud.dataflow.rest.client.VndErrorResponseErrorHandler.handleError(VndErrorResponseErrorHandler.java:65)
at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63)
at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:780)
at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:738)
at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:672)
at org.springframework.web.client.RestTemplate.postForObject(RestTemplate.java:416)
at org.springframework.cloud.dataflow.rest.client.StreamTemplate.updateStream(StreamTemplate.java:192)
at org.springframework.cloud.dataflow.shell.command.StreamCommands.updateStream(StreamCommands.java:242)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:282)
at org.springframework.shell.core.SimpleExecutionStrategy.invoke(SimpleExecutionStrategy.java:68)
at org.springframework.shell.core.SimpleExecutionStrategy.execute(SimpleExecutionStrategy.java:59)
at org.springframework.shell.core.AbstractShell.executeCommand(AbstractShell.java:134)
at org.springframework.shell.core.JLineShell.promptLoop(JLineShell.java:533)
at org.springframework.shell.core.JLineShell.run(JLineShell.java:179)
at java.lang.Thread.run(Thread.java:748)
```

skipper logs:
```
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT 2021-03-17 20:57:53.095 ERROR 50 — [eTaskExecutor-1] o.s.c.s.s.statemachine.AbstractAction : Action execution failed class=[class org.springframework.cloud.skipper.server.statemachine.UpgradeStartAction]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT org.springframework.cloud.skipper.SkipperException: Package to upgrade has no difference than existing deployed/deleted package. Not upgrading.
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.cloud.skipper.server.deployer.DefaultReleaseManager.createReport(DefaultReleaseManager.java:190) ~[spring-cloud-skipper-server-core-2.6.0.jar:2.6.0]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.cloud.skipper.server.service.ReleaseReportService.createReport(ReleaseReportService.java:118) ~[spring-cloud-skipper-server-core-2.6.0.jar:2.6.0]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.cloud.skipper.server.service.ReleaseReportService$$FastClassBySpringCGLIB$$6bf7cb0e.invoke(<generated>) ~[spring-cloud-skipper-server-core-2.6.0.jar:2.6.0]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.2.10.RELEASE.jar:5.2.10.RELEASE]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:771) ~[spring-aop-5.2.10.RELEASE.jar:5.2.10.RELEASE]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.2.10.RELEASE.jar:5.2.10.RELEASE]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:749) ~[spring-aop-5.2.10.RELEASE.jar:5.2.10.RELEASE]
2021-03-17T14:57:53.09-0600 [APP/PROC/WEB/0] OUT at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:367) ~[spring-tx-5.2.10.RELEASE.jar:5.2.10.RELEASE]
```

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Fix server error message to be included by setting explicit property
 - Add ReleaseUpgradeException to handle the exceptions thrown when performing the release upgrade
 - Add test

Resolves #997

### CODE REVIEW

**Comment by jvalkeal** [comment]
@ilayaperumalg you should handle milestoning and backport issues via originating issue.

File: spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/controller/SkipperErrorAttributes.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import java.util.Map;
 
 import org.springframework.boot.web.servlet.error.DefaultErrorAttributes;
+import org.springframework.boot.web.servlet.error.ErrorAttributes;
```
Comment (jvalkeal) [inline spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/controller/SkipperErrorAttributes.java:21]: Import probably not used

**Comment by jvalkeal** [comment]
Other than that, lgtm

File: spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/controller/SkipperErrorAttributes.java (hunk 1 above)
Comment (ilayaperumalg) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/controller/SkipperErrorAttributes.java:21]: This one I explicitly imported for the Javadoc comment section. Also, the build would have failed if this isn't used ;)

Comment (jvalkeal) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/controller/SkipperErrorAttributes.java:21]: Ah, javadoc error, not build error.

**Pull request merged**